### PR TITLE
Adding performance tests for DataFrame.groupby

### PR DESCRIPTION
**Pull request by kozlov-alexey**
Some perf test results (no scalability observed at all, need to investigate further):
```
[11:49:11]	[Step 3/3] DataFrame.groupby.min            1        Python 200000  0.009272  0.009256  0.009473       NaN       NaN
[11:49:11]	[Step 3/3]                                           SDC    200000  0.133266  0.132519  0.135123  2.788937  0.005203
[11:49:11]	[Step 3/3]                                  2        Python 200000  0.009156  0.009132  0.009373       NaN       NaN
[11:49:11]	[Step 3/3]                                           SDC    200000  0.130509  0.129958  0.130725  2.700916  0.005142
[11:49:11]	[Step 3/3]                                  4        Python 200000  0.009331  0.009276  0.009525       NaN       NaN
[11:49:11]	[Step 3/3]                                           SDC    200000  0.124665  0.124538  0.125711  2.722667  0.005059
[11:49:11]	[Step 3/3]                                  8        Python 200000  0.009249  0.009233  0.009411       NaN       NaN
[11:49:11]	[Step 3/3]                                           SDC    200000  0.127765  0.127684  0.127910  2.479266  0.004
```

### CODE REVIEW

**Comment by densmirn** [comment]
According to the report from CI:
`test_df_groupby_count_sort_false (sdc.tests.tests_perf.test_perf_df_groupby.TestDFGroupByMethods) ... ERROR`

**Comment by kozlov-alexey** [comment]
> According to the report from CI:
> `test_df_groupby_count_sort_false (sdc.tests.tests_perf.test_perf_df_groupby.TestDFGroupByMethods) ... ERROR`

That's very strange, as I re-run them and locally all the tests pass (including the one failed before):
----------------------------------------------------------------------
Ran 12 tests in 429.663s
OK

**Comment by kozlov-alexey** [comment]
@PokhodenkoSA, @AlexanderKalistratov 
With outer parfor removed compilation succeeds, so perf tests now pass: [TC-job](https://teamcity-or.intel.com/viewLog.html?buildId=4710284&buildTypeId=Sat_Sdc_Linux_SdcPy37numpy116_Testing_SdcBenchmarks&tab=buildLog)
Shall we merge this?

File: sdc/tests/tests_perf/test_perf_df_groupby.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+            record = base.copy()
+            record['test_type'] = 'SDC'
+            self._test_jitted(pyfunc, record, *args)
+            self.test_results.add(**record)
+
+            record = base.copy()
+            record['test_type'] = 'Python'
+            self._test_python(pyfunc, record, *args)
+            self.test_results.add(**record)
```
Comment (PokhodenkoSA) [inline sdc/tests/tests_perf/test_perf_df_groupby.py]: Consider using `_test_jit` and `_test_py` from `TestBase`.

**Comment by PokhodenkoSA** [comment]
> Shall we merge this?

Any implementation is better than nothing. I think we should merge it. 
But new perf tests reuse existing code very little (see other benchmarks) and reimplement in similar way many parts.

**Pull request merged**